src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -6,42 +6,36 @@
  * Side Public License, v 1.
  */
 
-import { cloneDeep, isEqual } from 'lodash';
 import { i18n } from '@kbn/i18n';
 import { History } from 'history';
-import { COMPARE_ALL_OPTIONS, compareFilters, Filter, FilterStateStore } from '@kbn/es-query';
 import {
   createKbnUrlStateStorage,
-  createStateContainer,
   IKbnUrlStateStorage,
-  ReduxLikeStateContainer,
   StateContainer,
-  syncState,
   withNotifyOnErrors,
 } from '@kbn/kibana-utils-plugin/public';
 import {
-  connectToQueryState,
   DataPublicPluginStart,
   FilterManager,
   QueryState,
   SearchSessionInfoProvider,
-  syncQueryStateWithUrl,
 } from '@kbn/data-plugin/public';
 import { DataView } from '@kbn/data-views-plugin/public';
 import { SavedSearch } from '@kbn/saved-search-plugin/public';
+import { loadDataView, resolveDataView } from '../utils/resolve_data_view';
 import { DataStateContainer, getDataStateContainer } from './discover_data_state_container';
 import { DiscoverSearchSessionManager } from './discover_search_session';
 import { DiscoverAppLocatorParams, DISCOVER_APP_LOCATOR } from '../../../../common';
-import { AppState } from './discover_app_state_container';
+import {
+  AppState,
+  DiscoverAppStateContainer,
+  getDiscoverAppStateContainer,
+} from './discover_app_state_container';
 import {
   getInternalStateContainer,
   InternalStateContainer,
 } from './discover_internal_state_container';
-import { getStateDefaults } from '../utils/get_state_defaults';
 import { DiscoverServices } from '../../../build_services';
-import { handleSourceColumnState } from '../../../utils/state_helpers';
-import { cleanupUrlState } from '../utils/cleanup_url_state';
-import { getValidFilters } from '../../../utils/get_valid_filters';
 
 export interface AppStateUrl extends Omit<AppState, 'sort'> {
```

Comment:
It looks like `AppStateUrl` also isn't used in `discover_state`, and it might make sense to move it to `discover_app_state_container` instead. If we moved `AppStateUrl` and `APP_STATE_URL_KEY` to `discover_app_state_container`, it would have no imports left from `discover_state` and would fix the circular dependency between `discover_app_state_container` and `discover_state`.

Reply:
good point! thx! done!